src/rsz/src/RepairSetup.cc

```diff
@@ -965,23 +1007,34 @@ RepairSetup::equivCellPins(const LibertyCell *cell, sta::LibertyPortSet &ports)
     int inputs = 0;
 
     // count number of output ports. Skip ports with > 1 output for now.
+    LibertyPort* output_port = nullptr;
     while (port_iter.hasNext()) {
         LibertyPort *port = port_iter.next();
         if (port->direction()->isOutput()) {
-            ++outputs;
+          output_port = port;
+          ++outputs;
         } else {
           ++inputs;
         }
     }
 
+    if (output_port == nullptr) {
+      return;
+    }
+
+    sta::FuncExpr* expr = output_port->function();
     if (outputs == 1 && inputs >= 2) {
         sta::LibertyCellPortIterator port_iter2(cell);
         while (port_iter2.hasNext()) {
-            LibertyPort *port = port_iter2.next();
-            sta::FuncExpr *expr = port->function();
-            if (expr != nullptr) {
-                getEquivPortList(expr, ports);
-            }
+          LibertyPort* candidate_port = port_iter2.next();
+          if (!candidate_port->direction()->isInput()) {
+            continue;
+          }
+
+          if (expr != nullptr && input_port != candidate_port
+              && isPortEqiv(expr, cell, input_port, candidate_port)) {
```

Comment:
Won't this lead to duplicate checks (eg A can be swapped with B implies B can be swapped with A)?

Reply:
Yeah, but that's an optimization, and this runs over the entire library in less than a second. No point in complicating it unless it's too slow.